src/main.rs

```diff
@@ -802,11 +812,50 @@ async fn handle_request(
     }
 }
 
+#[derive(Clone)]
+struct MetricsHandlerArgs {
+    exporter: PrometheusExporter,
+}
+
+async fn metrics_handler(
+    Extension(MetricsHandlerArgs { exporter }): Extension<MetricsHandlerArgs>,
+    _: Request<Body>,
+) -> Response<Body> {
+    let metric_families = exporter.registry().gather();
+
+    let encoder = TextEncoder::new();
+
+    let mut result = Vec::new();
+    if encoder.encode(&metric_families, &mut result).is_err() {
```

Comment:
Log the error here?

Reply:
@Daniel-Bloom-dfinity Are you OK with making this a future enhancement? It'll make more sense once we throw in some structured logging later.